### PR TITLE
Give different <title> tag for each page #3694

### LINKED ISSUE #3694: Give different <title> tag for each page
For example, all instructor pages currently have the same `<title>` tag `TEAMMATES - Instructor`. It can be made specific for each page.

### PR DESCRIPTION
**Pull request by amarlearning**
Fixes #3694 

- Changed the web page title to `<heading> [Instructor/Student/Administrator]`

### CODE REVIEW

**Comment by amarlearning** [comment]
ready for review

File: src/main/webapp/jsp/adminSessions.jsp [hunk 1]
```diff
@@ -9,7 +9,7 @@
 <c:set var="jsIncludes">
   <script type="text/javascript" src="/js/adminSessions.js"></script>
 </c:set>
-<ta:adminPage bodyTitle="Ongoing Sessions" pageTitle="TEAMMATES - Administrator Sessions" cssIncludes="${cssIncludes}" jsIncludes="${jsIncludes}">
+<ta:adminPage bodyTitle="Ongoing Sessions" pageTitle="Ongoing Sessions [Administrator] Sessions" cssIncludes="${cssIncludes}" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/adminSessions.jsp]: `Sessions` is repeated.

**Comment by amarlearning** [comment]
ready for review

File: src/main/webapp/WEB-INF/tags/student/studentPage.tag [hunk 2]
```diff
@@ -15,7 +15,7 @@
   <t:bodyFooter />
 </c:set>
 
-<t:page pageTitle="${pageTitle}" bodyTitle="${bodyTitle}">
+<t:page pageTitle="${pageTitle} [Student]" bodyTitle="${bodyTitle}">
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/student/studentPage.tag]: Perhaps student pages don't need a suffix at all, as that is probably the most common usage (in terms of traffic) of TEAMMATES. Would @damithc be agreeable?

File: src/main/webapp/jsp/instructorCourseJoinConfirmation.jsp [hunk 3]
```diff
@@ -7,7 +7,7 @@
 <c:set var="jsIncludes">
   <script type="text/javascript" src="/js/instructorCourseJoinConfirmation.js"></script>
 </c:set>
-<ti:instructorPage pageTitle="TEAMMATES - Instructor" bodyTitle="" jsIncludes="${jsIncludes}">
+<ti:instructorPage pageTitle="Course Confirmation" bodyTitle="" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorCourseJoinConfirmation.jsp]: `Course Join Confirmation`?

File: src/main/webapp/jsp/studentCourseDetails.jsp [hunk 4]
```diff
@@ -9,7 +9,7 @@
   <script type="text/javascript" src="/js/studentCourseDetails.js"></script>
 </c:set>
 
-<ts:studentPage bodyTitle="Team Details for ${data.studentCourseDetailsPanel.courseId}" pageTitle="TEAMMATES - Student" jsIncludes="${jsIncludes}">
+<ts:studentPage bodyTitle="Team Details for ${data.studentCourseDetailsPanel.courseId}" pageTitle="Team Details" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/studentCourseDetails.jsp]: Just highlighting that if we drop the `pageTitle` attribute, the title would include the course ID.

File: src/main/webapp/jsp/studentCourseJoinConfirmation.jsp [hunk 5]
```diff
@@ -6,7 +6,7 @@
 <c:set var="jsIncludes">
   <script type="text/javascript" src="/js/studentCourseJoinConfirmation.js"></script>
 </c:set>
-<ts:studentPage pageTitle="TEAMMATES - Student" bodyTitle="" jsIncludes="${jsIncludes}">
+<ts:studentPage pageTitle="Course Confirmation" bodyTitle="" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/studentCourseJoinConfirmation.jsp]: `Course Join Confirmation`?

File: src/main/webapp/jsp/instructorCourseEnroll.jsp [hunk 6]
```diff
@@ -15,7 +15,7 @@
   <%=Const.ParamsNames.SESSION_TOKEN%>
 </c:set>
 
-<ti:instructorPage pageTitle="TEAMMATES - Instructor" bodyTitle="Enroll Students for ${data.courseId}" cssIncludes="${cssIncludes}" jsIncludes="${jsIncludes}">
+<ti:instructorPage pageTitle="Enroll Students" bodyTitle="Enroll Students for ${data.courseId}" cssIncludes="${cssIncludes}" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorCourseEnroll.jsp]: Just highlighting that if we drop the `pageTitle` attribute, the title would include the course ID.

File: src/main/webapp/jsp/instructorCourseEnrollResult.jsp [hunk 7]
```diff
@@ -17,7 +17,7 @@
   <%=Const.ParamsNames.STUDENTS_ENROLLMENT_INFO%>
 </c:set>
 
-<ti:instructorPage pageTitle="TEAMMATES - Instructor" bodyTitle="Enrollment Results for ${data.courseId}" jsIncludes="${jsIncludes}">
+<ti:instructorPage pageTitle="Enrollment Results" bodyTitle="Enrollment Results for ${data.courseId}" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorCourseEnrollResult.jsp]: Just highlighting that if we drop the `pageTitle` attribute, the title would include the course ID.

File: src/main/webapp/jsp/instructorFeedbackResultsByGiverQuestionRecipient.jsp [hunk 8]
```diff
@@ -6,7 +6,7 @@
   <script type="text/javascript" src="/js/instructorFeedbackResultsGQR.js"></script>
 </c:set>
 
-<results:resultsPage pageTitle="TEAMMATES - Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
+<results:resultsPage pageTitle="Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorFeedbackResultsByGiverQuestionRecipient.jsp]: Just highlighting that if we drop the `pageTitle` attribute, the title would simply be `Session Results [Instructor]`.

File: src/main/webapp/jsp/instructorFeedbackResultsByGiverRecipientQuestion.jsp [hunk 9]
```diff
@@ -7,6 +7,6 @@
   <script type="text/javascript" src="/js/instructorFeedbackResultsGRQ.js"></script>
 </c:set>
 
-<results:resultsPage pageTitle="TEAMMATES - Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
+<results:resultsPage pageTitle="Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorFeedbackResultsByGiverRecipientQuestion.jsp]: Same here.

File: src/main/webapp/jsp/instructorFeedbackResultsByQuestion.jsp [hunk 10]
```diff
@@ -6,7 +6,7 @@
   <script type="text/javascript" src="/js/instructorFeedbackResultsQuestion.js"></script>
 </c:set>
 
-<results:resultsPage pageTitle="TEAMMATES - Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
+<results:resultsPage pageTitle="Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorFeedbackResultsByQuestion.jsp]: Same here.

File: src/main/webapp/jsp/instructorFeedbackResultsByRecipientGiverQuestion.jsp [hunk 11]
```diff
@@ -7,6 +7,6 @@
   <script type="text/javascript" src="/js/instructorFeedbackResultsRGQ.js"></script>
 </c:set>
 
-<results:resultsPage pageTitle="TEAMMATES - Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
+<results:resultsPage pageTitle="Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorFeedbackResultsByRecipientGiverQuestion.jsp]: Same here.

File: src/main/webapp/jsp/instructorFeedbackResultsByRecipientQuestionGiver.jsp [hunk 12]
```diff
@@ -6,6 +6,6 @@
   <script type="text/javascript" src="/js/instructorFeedbackResultsRQG.js"></script>
 </c:set>
 
-<results:resultsPage pageTitle="TEAMMATES - Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
+<results:resultsPage pageTitle="Feedback Session Results" bodyTitle="Session Results" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorFeedbackResultsByRecipientQuestionGiver.jsp]: Same here.

File: src/main/webapp/jsp/instructorHome.jsp [hunk 13]
```diff
@@ -6,7 +6,7 @@
 <c:set var="jsIncludes">
   <script type="text/javascript" src="/js/instructorHome.js"></script>
 </c:set>
-<ti:instructorPage pageTitle="TEAMMATES - Instructor" bodyTitle="Instructor Home" jsIncludes="${jsIncludes}">
+<ti:instructorPage pageTitle="Home" bodyTitle="Instructor Home" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorHome.jsp]: Just highlighting that if we drop the `pageTitle` attribute, the `bodyTitle` should just be `Home` as the title would already read `Home [Instructor]`.

File: src/main/webapp/jsp/instructorStudentRecords.jsp [hunk 14]
```diff
@@ -10,7 +10,7 @@
   <script type="text/javascript" src="/js/instructorStudentRecords.js"></script>
 </c:set>
 <c:set var="bodyTitle">${fn:escapeXml(data.studentName)}'s Records<small class="muted"> - ${data.courseId}</small></c:set>
-<ti:instructorPage pageTitle="TEAMMATES - Instructor" jsIncludes="${jsIncludes}" bodyTitle="${bodyTitle}">
+<ti:instructorPage pageTitle="Student Records" jsIncludes="${jsIncludes}" bodyTitle="${bodyTitle}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/instructorStudentRecords.jsp]: Just highlighting that if we drop the `pageTitle` attribute, the title would include the student's name.

File: src/main/webapp/jsp/studentFeedbackResults.jsp [hunk 15]
```diff
@@ -8,7 +8,7 @@
   <script type="text/javascript" src="/js/studentFeedbackResults.js"></script>
 </c:set>
 
-<ts:studentPage bodyTitle="Feedback Results - Student" pageTitle="TEAMMATES - Feedback Results" jsIncludes="${jsIncludes}">
+<ts:studentPage bodyTitle="Feedback Results - Student" pageTitle="Feedback Results" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/studentFeedbackResults.jsp]: I feel the `- Student` suffix can be dropped.

File: src/main/webapp/jsp/studentHome.jsp [hunk 16]
```diff
@@ -7,7 +7,7 @@
 <c:set var="jsIncludes">
   <script type="text/javascript" src="/js/studentHome.js"></script>
 </c:set>
-<ts:studentPage pageTitle="TEAMMATES - Student" bodyTitle="Student Home" jsIncludes="${jsIncludes}">
+<ts:studentPage pageTitle="Home" bodyTitle="Student Home" jsIncludes="${jsIncludes}">
```
Comment (whipermr5) [inline src/main/webapp/jsp/studentHome.jsp]: Just highlighting that if we drop the `pageTitle` attribute, the title would also be `Student Home`.

File: src/main/webapp/WEB-INF/tags/student/studentPage.tag (hunk 2 above)
Comment (damithc) [inline src/main/webapp/WEB-INF/tags/student/studentPage.tag]: Yes, I'm ok with that

**Comment by damithc** [comment]
> Would @damithc be agreeable to making the pageTitle always follow the bodyTitle?

Yes I'm ok with that. It's a good idea to reduce duplication.

File: src/main/webapp/jsp/instructorCourseEnroll.jsp (hunk 6 above)
Comment (amarlearning) [inline src/main/webapp/jsp/instructorCourseEnroll.jsp]: @whipermr5 are you sure about making `course ID` as the title of the page?

Comment (whipermr5) [inline src/main/webapp/jsp/instructorCourseEnroll.jsp]: You have reservations?

File: src/main/webapp/jsp/studentHome.jsp (hunk 16 above)
Comment (amarlearning) [inline src/main/webapp/jsp/studentHome.jsp]: @whipermr5 I think we should be consistent over here. In instructors home page we are using `Home [Instructor]` and here we are using `Student Home` :thinking:

File: src/main/webapp/jsp/instructorCourseEnroll.jsp (hunk 6 above)
Comment (amarlearning) [inline src/main/webapp/jsp/instructorCourseEnroll.jsp]: Sorry, I misread that :+1:

File: src/main/webapp/jsp/studentHome.jsp (hunk 16 above)
Comment (whipermr5) [inline src/main/webapp/jsp/studentHome.jsp]: I feel that it's mainly caused by student being a special category where we don't add the suffix. As long as `Home [Admin]` and `Home [Instructor]` are consistent, I feel it's fine.

File: src/main/webapp/jsp/instructorCourseEnroll.jsp (hunk 6 above)
Comment (whipermr5) [inline src/main/webapp/jsp/instructorCourseEnroll.jsp]: Oh haha I see - yup I meant the actual course ID and not the string literal `course ID`.

**Comment by amarlearning** [comment]
>  dropping the `pageTitle` attribute from `adminPage`, `instructorPage` and `studentPage` completely

@whipermr5 Just confirming before making any changes, you want to remove `pageTitle` attribute from `AdminPage`, `InstructorPage` and `studentPage`?
```java
<t:page pageTitle="${pageTitle} [Administrator]" bodyTitle="${bodyTitle}">
```
**to**
```java
<t:page bodyTitle="${bodyTitle}">
```
***
If yes, then probably we would have to make changes inside `page.tag` file
```html
<title>${pageTitle}</title>
```
**to**
```html
<title>${bodyTitle}</title>
```
***
If this is the case, can you tell me how would I achive [this](https://github.com/TEAMMATES/teammates/pull/7806#discussion_r129232430). Please correct me if i am wrong somewhere :sweat_smile:

**Comment by whipermr5** [comment]
@amarlearning Don't edit `page.tag`. Instead, use:
```jsp
<t:page pageTitle="${bodyTitle} [Administrator]" bodyTitle="${bodyTitle}">
```

And delete the `pageTitle` attribute in `adminPage.tag`. Also, it'd be good to rename the `bodyTitle` attribute in `adminPage.tag` to `title`, resulting in:
```jsp
<t:page pageTitle="${title} [Administrator]" bodyTitle="${title}">
```

Do the same for `instructorPage` and `studentPage` (but leave the suffix out for `studentPage`).

**Comment by amarlearning** [comment]
ready for review

**Pull request merged**